patches/server/0856-Ability-to-control-player-s-insomnia-and-phantoms.patch

```diff
@@ -0,0 +1,123 @@
++    }
++
++    public int phantomsSpawnAttemptMinSeconds = 60;
++    public int phantomsSpawnAttemptMaxSeconds = 119;
++    public boolean phantomSpawnerEnabled = false;
++    private void phantomsSpawnAttemptSettings() {
++        phantomsSpawnAttemptMinSeconds = getInt("game-mechanics.phantoms-spawn-attempt-seconds.min", phantomsSpawnAttemptMinSeconds);
++        phantomsSpawnAttemptMaxSeconds = getInt("game-mechanics.phantoms-spawn-attempt-seconds.max", phantomsSpawnAttemptMaxSeconds);
++
++        if (phantomsSpawnAttemptMinSeconds > phantomsSpawnAttemptMaxSeconds) {
++            int temp = phantomsSpawnAttemptMinSeconds;
++            phantomsSpawnAttemptMinSeconds = phantomsSpawnAttemptMaxSeconds;
++            phantomsSpawnAttemptMaxSeconds = temp;
++        }
++
++        phantomSpawnerEnabled = (phantomsSpawnAttemptMinSeconds >= 0 && phantomsSpawnAttemptMaxSeconds >= 0) && (phantomsSpawnAttemptMinSeconds != 0 || phantomsSpawnAttemptMaxSeconds != 0);
++    }
++
+     public boolean zombiesAlwaysCanPickUpLoot;
+     public boolean skeletonsAlwaysCanPickUpLoot;
+     private void setMobsAlwaysCanPickUpLoot() {
+diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
+index a2abb8aa1a257ccd2b5dbddc037fffc6eb600758..10e501eae443d0239c7b67c8643673c9e491502b 100644
+--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
++++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
+@@ -2,6 +2,7 @@ package net.minecraft.server.level;
+ 
+ import com.google.common.annotations.VisibleForTesting;
+ import co.aikar.timings.TimingHistory; // Paper
++import com.google.common.collect.ImmutableList; // Paper
+ import com.google.common.collect.Lists;
+ import com.mojang.datafixers.DataFixer;
+ import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+@@ -129,6 +130,7 @@ import net.minecraft.world.level.gameevent.GameEvent;
+ import net.minecraft.world.level.gameevent.GameEventListenerRegistrar;
+ import net.minecraft.world.level.gameevent.vibrations.VibrationPath;
+ import net.minecraft.world.level.levelgen.Heightmap;
++import net.minecraft.world.level.levelgen.PhantomSpawner; // Paper
+ import net.minecraft.world.level.levelgen.feature.StructureFeature;
+ import net.minecraft.world.level.levelgen.structure.BoundingBox;
+ import net.minecraft.world.level.levelgen.structure.StructureCheck;
+@@ -495,7 +497,13 @@ public class ServerLevel extends Level implements WorldGenLevel {
+         this.dragonParts = new Int2ObjectOpenHashMap();
+         this.tickTime = flag1;
+         this.server = minecraftserver;
+-        this.customSpawners = list;
++        // Paper start
++        if (this.paperConfig.phantomSpawnerEnabled) {
++            this.customSpawners = list;
++        } else {
++            this.customSpawners = ImmutableList.copyOf(list.stream().filter(s -> !(s instanceof PhantomSpawner)).toList());
```

Comment:
Can just use List.copyOf

Also make *sure* the list of custom spawners parameter in the ServerLevel ctor isn't used before this line.

Reply:
I don't think it is used before this code. Did you spot something?

Reply:
No, I just didn't have the code open to check, and was making sure you checked. so if there isn't then its good.